Mathlib/Data/Array/extractLemmas.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2024 Jiecheng Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiecheng Zhao
+-/
+import Std.Data.Array.Lemmas
+/-!
+# lemmas about Array.extract
+
+Some useful lemmas about Array.extract
+-/
+set_option autoImplicit true
+
+namespace Array
+
+
+@[simp]
+theorem extract_zero_length {a : Array α} :
+    a.extract i i = #[] := by
+  refine extract_empty_of_stop_le_start a ?h
+  exact Nat.le_refl i
+
+theorem size_extract_to_stop {a: Array α} : (e ≤ a.size) → (a.extract s e).size = e - s := by
+  intro h1
+  simp only [size_extract]
+  rw [Nat.min_eq_left h1]
+
+theorem extract_append_left {a b: Array α} {i j : Nat} (h: j ≤ a.size):
+    (a ++ b).extract i j = a.extract i j := by
+  apply ext
+  · have h' : j ≤ a.size + b.size := by
+      apply Nat.le_trans h
+      exact Nat.le_add_right (size a) (size b)
+    simp only [size_extract, size_append, h', Nat.min_eq_left, h]
+  · intro h1 h2 h3
+    simp only [get_extract]
+    rw [get_append_left]
+
+theorem extract_append_right {a b: Array α} {i j : Nat} (h: i ≥ a.size):
+    (a ++ b).extract i j = b.extract (i - a.size) (j - a.size) := by
```

Comment:
I'd hope there's an easier argument here, based on comparing the `k`-th element of each side. Do we have the necessary lemmas about `getElem` and `extract`?